hlapipeline/alignimages.py

```diff
@@ -0,0 +1,112 @@
+                                 "sbc":
+                                     {"fwhmpsf":0.065, # TODO: Verify value
+                                      "platescale":0.03}, # TODO: hla DGSL pipleine value: 0.030. Instrument handbook value: 0.032, and later ~0.034 × 0.030"/pixel Verify correct value.
+                                 "wfc":{"fwhmpsf":0.076, # TODO: Verify value
+                                      "platescale":0.05}}, # TODO: nothing. pipeline value in agreement with instrument handbook value.
+                            "wfc3":
+                                {"ir":{"fwhmpsf":0.14, # TODO: nothing. pipeline value in agreement with instrument handbook value.
+                                      "platescale":0.09}, # TODO: hla DGSL pipleine value: 0.09. Instrument handbook value: 0.13, also later 0.135×0.121. Verify correct value.
+                                 "uvis":{"fwhmpsf":0.076, # TODO: nothing. pipeline value in agreement with instrument handbook value.
+                                      "platescale":0.04}}} # TODO: hla DGSL pipleine value: 0.04. Instrument handbook value: 0.04, but also later 0.0395×0.0395. Verify correct value.
+#-----------------------------------------------------------------------------------------------------------------------
+def main(imgList):
+    """
+    Main calling function.
+
+    :param imgList: list of images
+    :type imgList: list
+    :return: nothing for now.
+    """
+
+    # 1: Interpret input data and optional parameters
+
+    # 2: Apply filter to input observations to insure that they meet minimum criteria for being able to be aligned
+
+    # 3: Build WCS for full set of input observations
+    refwcs = amutils.build_reference_wcs(imgList)
+
+    # 4: Retrieve list of astrometric sources from database
+
+    # 5: Extract catalog of observable sources from each input image
+    extracted_sources = generate_source_catalogs(imgList,refwcs)
+
+    # 6: Cross-match source catalog with astrometric reference source catalog
+
+    # 7: Perform fit between source catalog and reference catalog
+#-----------------------------------------------------------------------------------------------------------------------
+def generate_source_catalogs(imgList,refwcs):
+    """
+    Genreates a dictionary of source catalogs keyed by image name.
+
+    :param imgList:
+    :return:
+    """
+    sourceCatalogDict = {}
+
+    for imgName in imgList:
+        print("Image name:           ", imgName)
+
+        sourceCatalogDict[imgName] = {}
+
+        # open image
+        imgHDU = fits.open(imgName)
+        imgPrimaryHeader = imgHDU[0].header
+        instrument = imgPrimaryHeader['INSTRUME'].lower()
+        detector = imgPrimaryHeader['DETECTOR'].lower()
+
+        # get instrument/detector-specific image alignment parameters
+        sourceCatalogDict[imgName]["params"] = detector_specific_params[instrument][detector]
+
```

Comment:
Consider reading in HSTWCS object for each image and using that to define the 'plate scale' used to convert the fwhm from arcseconds to pixels.

Reply:
suggestions implemented in commit [946f945](https://github.com/dulude/hlapipeline/commit/946f94521ebebf13c775f5125ddf93bab0fda16a)